package.json

```diff
@@ -25,29 +26,39 @@
     "async"
   ],
   "scripts": {
-    "build-dist": "mkdir -p dist && rollup src/main.js | babel -m umd --module-id creed -o dist/creed.js --loose es6.classes",
+    "compile": "npm run compile-src && npm run compile-test",
+    "compile-src": "mkdirp build/src && buble -m -i src -o build/src --no modules",
+    "compile-test": "mkdirp build/test && buble -m -i test -o build/test --no modules,generator --target node:4",
+    "build-dist": "npm run compile && mkdirp dist && rollup -c",
     "build": "npm run build-dist && uglifyjs -c 'warnings=false' -m -o dist/creed.min.js -- dist/creed.js",
     "preversion": "npm run build",
-    "test-aplus": "promises-aplus-tests test/aplus.js --reporter dot",
-    "test-main": "istanbul cover --root src _mocha -- --reporter dot",
     "check-coverage": "istanbul check-coverage --statements 100 --branches 100 --lines 100 --functions 100 coverage/coverage*.json",
     "lint": "jsinspect src && eslint src",
-    "test": "npm run lint && npm run build && npm run test-main && npm run check-coverage && npm run test-aplus"
+    "pretest": "npm run lint && npm run build",
+    "test": "istanbul cover --root build/src _mocha -- build/test/*-test.js --require babel-register",
+    "posttest": "npm run test-aplus",
+    "test-aplus": "promises-aplus-tests test/aplus.js --reporter dot"
```

Comment:
Yeah, it's getting pretty `npm run` heavy.  I'm hoping that using @davidchase's [buba](https://www.npmjs.com/package/buba) will reduce the gymnastics at least a little.